src/main/java/hlf/java/rest/client/service/impl/ChaincodeOperationsServiceImpl.java

```diff
@@ -94,26 +96,29 @@ public String getCurrentSequence(
           queryChaincodeDefinitionProposalResponses =
               channel.lifecycleQueryChaincodeDefinition(
                   queryLifecycleQueryChaincodeDefinitionRequest, peers);
-      Set<Long> sequenceNumbers = new HashSet<>();
+      long sequence = -1L;
       for (LifecycleQueryChaincodeDefinitionProposalResponse response :
           queryChaincodeDefinitionProposalResponses) {
-        if (response.getVersion().equals(chaincodeVersion)) {
-          sequenceNumbers.add(response.getSequence());
+        if (response.getStatus() == ProposalResponse.Status.SUCCESS
+            && response.getVersion().equals(chaincodeVersion)) {
+          sequence = response.getSequence();
+          break;
+        } else {
+          if (404 == response.getChaincodeActionResponseStatus()) {
+            // not found set sequence to 1;
+            sequence = 1;
+            break;
+          }
```

Comment:
[nitpick] : Use constant HttpStatus.NOT_FOUND
Q: can there be other scenerio like with status code 500?

Reply:
1. Constant added.
2. We only what to check for scenario where given chaincode name isn't found i.e 404.